### PR TITLE
Be_between matcher

### PR DESCRIPTION
**Pull request by Pazus**
Be_between matcher implementation.
Tests are included.

On review please pay attention to the null processing, not sure I've implemented it correctly as designed. What if expected value is null, what if actual is null.

### CODE REVIEW

**Comment by Pazus** [comment]
I'll come back and fix the issue

**Comment by jgebal** [comment]
Can you please add support for varchar2 as well (for completeness)?
This code is perfectly valid in PLSQL so maybe asserts should also support it:
```sql
begin
  if 'b' between 'a' and 'cb' then
    dbms_output.put_line('ok');
  end if;
end;
/
```

File: tests/asssertions/ut.expect.to_be_between.GivesSuccessWhenDifferentTypes.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+PROMPT Gives failure when expected are numbers and and actual is varchar
```
Comment (jgebal) [inline tests/asssertions/ut.expect.to_be_between.GivesSuccessWhenDifferentTypes.sql]: the prompt is not correct.

File: tests/asssertions/ut.expect.to_be_between.GivesSuccessWhenDifferentTypes.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+@@asssertions/common/ut.expect.to_be_between.scalar.different_types.common.sql 'varchar2(4000)' 'number' '''1''' '0' '2'
+
+PROMPT Gives failure when expected are varchars and and actual is number
```
Comment (jgebal) [inline tests/asssertions/ut.expect.to_be_between.GivesSuccessWhenDifferentTypes.sql]: and here

File: source/expectations/matchers/be_between.tpb [hunk 3]
```diff
@@ -58,6 +64,14 @@ create or replace type body be_between is
       begin
         l_result := l_actual.datavalue between l_lower.datavalue and l_upper.datavalue;
       end;
+    elsif self.lower_bound is of(ut_data_value_varchar2) and self.lower_bound is of(ut_data_value_varchar2) and a_actual is of(ut_data_value_varchar2) then
```
Comment (jgebal) [inline source/expectations/matchers/be_between.tpb]: typo lower_bound is checked twice and upper_bound is not checked.

File: source/expectations/ut_expectation_varchar2.tps [hunk 4]
```diff
@@ -1,6 +1,7 @@
 create or replace type ut_expectation_varchar2 under ut_expectation
 (
   overriding member procedure to_equal(self in ut_expectation_varchar2, a_expected varchar2, a_nulls_are_equal boolean := null),
+  member procedure to_be_between(self in ut_expectation_varchar2, a_lower_bound varchar2, a_higher_bound varchar2),
```
Comment (jgebal) [inline source/expectations/ut_expectation_varchar2.tps]: maybe we can use `upper_bound` not `higher_bound`

Comment (Pazus) [inline source/expectations/ut_expectation_varchar2.tps]: done

File: source/expectations/matchers/be_between.tpb (hunk 3 above)
Comment (Pazus) [inline source/expectations/matchers/be_between.tpb]: fixed!

**Pull request merged**